core/parachain/validator/impl/parachain_processor.cpp

```diff
@@ -1346,19 +1357,53 @@ namespace kagome::parachain {
     auto se = pm_->getStreamEngine();
     BOOST_ASSERT(se);
 
+    // TODO(turuslan): #1757, LargeStatement
+    auto message = std::make_shared<
+        network::WireMessage<network::ValidatorProtocolMessage>>(
+        network::ValidatorProtocolMessage{
+            network::StatementDistributionMessage{network::Seconded{
+                .relay_parent = relay_parent, .statement = statement}}});
+
+    std::unordered_set<network::PeerId> group_set;
+    if (auto r = runtime_info_->get_session_info(relay_parent)) {
+      auto &[session, info] = r.value();
+      if (info.our_group) {
+        for (auto &i : session.validator_groups[*info.our_group]) {
+          if (auto peer = query_audi_->get(session.discovery_keys[i])) {
+            group_set.emplace(peer->id);
+          }
+        }
+      }
+    }
+    std::vector<network::PeerId> group, any;
+    auto protocol = router_->getValidationProtocol();
+    se->forEachPeer(protocol, [&](const network::PeerId &peer) {
+      (group_set.count(peer) != 0 ? group : any).emplace_back(peer);
+    });
+    auto lucky = kMinGossipPeers - std::min(group.size(), kMinGossipPeers);
+    if (lucky != 0) {
+      std::shuffle(any.begin(), any.end(), random_);
+      any.erase(any.begin() + std::min(any.size(), lucky), any.end());
+    } else {
+      any.clear();
+    }
+
     logger_->trace(
         "Broadcasting StatementDistributionMessage.(relay_parent={}, validator "
         "index={}, sig={})",
         relay_parent,
         statement.payload.ix,
         statement.signature);
-    se->broadcast(
-        router_->getValidationProtocol(),
-        std::make_shared<
-            network::WireMessage<network::ValidatorProtocolMessage>>(
-            network::ValidatorProtocolMessage{
-                network::StatementDistributionMessage{network::Seconded{
-                    .relay_parent = relay_parent, .statement = statement}}}));
+
+    auto send = [&](const network::PeerId &peer) {
```

Comment:
Why extra lambda? Just easy
```
for (...)
   se->send(peer, protocol, message);

Reply:
This is used to deduplicate `seen.emplace({peer, message})`